src/AutoMapper/MappingEngine.cs

```diff
@@ -212,28 +190,22 @@ public TDestination DynamicMap<TDestination>(object source)
 
         public object DynamicMap(object source, Type sourceType, Type destinationType)
         {
+            ConfigurationProvider.CreateMissingTypeMaps = true;

```

Comment:
Oh and with DynamicMap being obsolete, do you not bother with this issues?
```
public void Main()
{
     Mapper.DynamicMap<A, NotMappedToA>(new A());
     Mapper.Map<B, NotMappedToB>(new B());
}
```
It's expected to throw a map not found exception, but with this commit it won't. and map B to NotMappedToB, since you don't set CreateMissingTypeMaps back to false after mapping